test/test.js

```diff
@@ -69,33 +69,30 @@ describe('onclickoutside hoc', function() {
 
 
   it('should call the specified handler when clicking the document', function() {
-    var Component = React.createClass({
-      getInitialState: function() {
-        return {
+    class Component extends React.Component {
```

Comment:
But (still thinking out loud) I don't think it makes sense to auto bind on the instance on stateless functions. The whole point of those is to be... stateless. So the logic here would be:
- If it has been created by inheritance: re-bind.
- If it has been created with createClass: binding already taken care of, do nothing.
- If it it a stateless method: binding shouldn't be necessary, do nothing.

But to test that logic, I do need to use es6 classes, unless I try and mock the inheritance, but that seems risky and more complex.